cpp/arcticdb/async/tasks.hpp

```diff
@@ -492,24 +492,18 @@ struct DecodeMetadataAndDescriptorTask : BaseTask {
             );
     }
 };
-
-template<typename ConstVarKeyGetter,
-        typename=std::enable_if_t<!std::is_reference_v<ConstVarKeyGetter> // Make obvious if iterator& is captured
-                        && std::is_same_v<decltype(*std::declval<ConstVarKeyGetter>()), const entity::VariantKey&>>>
 struct KeyExistsTask : BaseTask {
-    ConstVarKeyGetter key_;
+    const VariantKey key_;
     std::shared_ptr<storage::Library> lib_;
 
-    /**
-     * @param key Could be a vector iterator or a pointer to a const VariantKey.
-     */
-    KeyExistsTask(ConstVarKeyGetter key, std::shared_ptr<storage::Library> lib): key_(key), lib_(std::move(lib)) {
-        ARCTICDB_DEBUG(log::storage(), "Creating key exists task for key {}", variant_key_view(*key_));
+    KeyExistsTask(auto &&key, std::shared_ptr<storage::Library> lib): key_(std::forward<decltype(key)>(key)), lib_(std::move(lib)) {
```

Comment:
The advantage of `auto&&` is eliminated by the fact that all of the uses of this ctor pass by `const&`. Please go through the usages of `key_exists`, `key_exists_sync`, and `batch_key_exists` (including in the enterprise repo) and see if there are any usages which could take advantage of move semantics.

Reply:
I have just done another sweep and replace as much as I can.